tests/test_auth_requests_store.py

```
@@ -0,0 +1,239 @@
+async def test_set_decision_unknown_id_returns_none(tmp_path):
+    s = await _store(tmp_path)
+    result = await s.set_decision("no-such-id", "accepted", decided_by="admin")
+    assert result is None
+    await s.close()
+
+
+@pytest.mark.asyncio
+async def test_set_decision_concurrent_winner(tmp_path):
+    s = await _store(tmp_path)
+    rec = await s.create(identity_claim="a", framework="f", requested_scopes=[])
+    r1 = await s.set_decision(rec["id"], "accepted", canonical_id="c1", token="t1", decided_by="one")
```

Ruff (0.15.17)

[error] Possible hardcoded password assigned to argument: "token"

(S106)